.github/workflows/publish-docker-image.yml

```diff
@@ -0,0 +1,25 @@
+name: Publish Docker Image
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  publish-docker-image:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: docker image - quay.io
+        uses: aevea/action-kaniko@master
+        with:
+          registry: quay.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.QUAY_PASS }}
+          image: ${{ github.repository }}
+          build_file: Dockerfile
+          tag_with_latest: true
+          cache: true
+          cache_registry: quay.io/${{ github.repository_owner }}/cache
```

Comment:
Just a quick note: with this setup, Kaniko can ONLY build one `arch-based` image at a time.

Reply:
OK, noted, then github matrix is needed here